Checks/PWR003/example.f90

```diff
@@ -1,23 +1,25 @@
 ! PWR003: Explicitly declare pure functions
 
-module example_module
-   implicit none
-   integer :: c = 1
-contains
-   ! Depends on external data (c)
-   ! No side effects
-   pure function example_pure(a) result(res)
-      integer, intent(in) :: a
-      integer :: res
-      res = a + c
-   end function example_pure
+! Returns the gravity on Earth in m/s2
+function gravity()
+  use iso_c_binding, only : c_double
+  implicit none
+  real(kind=c_double) :: gravity
+  gravity = 9.8
+end function gravity
 
-   ! Depends on external data (c)
-   ! Modifies external data (c)
-   function example_impure(a) result(res)
-      integer, intent(in) :: a
-      integer :: res
-      c = c + 1
-      res = a + c
-   end function example_impure
-end module example_module
+! Computes the weight of each object in a vector
+subroutine example_f(n, M, W) bind(c)
+  use iso_c_binding, only : c_int, c_double
+
+  implicit none
+  integer(kind=c_int) :: i
+  integer(kind=c_int), intent(in), value :: n
+  real(kind=c_double) :: gravity
+  real(kind=c_double), dimension(1:n), intent(in) :: M
```

Comment:
It's not necessary to declare this `gravity` variable, or am I missing something?

Reply:
This is necessary because the functions are not enclosed within a module.

Reply:
True that, thanks for the clarification!